server/src/main/java/org/opensearch/index/remote/RemoteStoreEnums.java

```diff
@@ -0,0 +1,187 @@
+    public enum DataType {
+        DATA("data"),
+        METADATA("metadata"),
+        LOCK_FILES("lock_files");
+
+        private final String name;
+
+        DataType(String name) {
+            this.name = name;
+        }
+
+        public String getName() {
+            return name;
+        }
+    }
+
+    /**
+     * Enumerates the types of remote store paths resolution techniques supported by OpenSearch.
+     * For more information, see <a href="https://github.com/opensearch-project/OpenSearch/issues/12567">Github issue #12567</a>.
+     */
+    @PublicApi(since = "2.14.0")
+    public enum PathType {
+        FIXED {
+            @Override
+            public BlobPath generatePath(PathInput pathInput, PathHashAlgorithm hashAlgorithm) {
+                // Hash algorithm is not used in FIXED path type
+                return pathInput.basePath()
+                    .add(pathInput.indexUUID())
+                    .add(pathInput.shardId())
+                    .add(pathInput.dataCategory().getName())
+                    .add(pathInput.dataType().getName());
+            }
+
+            @Override
+            boolean requiresHashAlgorithm() {
+                return false;
+            }
+        },
+        HASHED_PREFIX {
+            @Override
+            public BlobPath generatePath(PathInput pathInput, PathHashAlgorithm hashAlgorithm) {
+                // TODO - We need to implement this, keeping the same path as Fixed for sake of multiple tests that can fail otherwise.
+                // throw new UnsupportedOperationException("Not implemented"); --> Not using this for unblocking couple of tests.
+                return pathInput.basePath()
+                    .add(pathInput.indexUUID())
+                    .add(pathInput.shardId())
+                    .add(pathInput.dataCategory().getName())
+                    .add(pathInput.dataType().getName());
+            }
+
+            @Override
+            boolean requiresHashAlgorithm() {
+                return true;
+            }
+        };
+
+        /**
+         * This method generates the path for the given path input which constitutes multiple fields and characteristics
+         * of the data.
+         *
+         * @param pathInput     input.
+         * @param hashAlgorithm hashing algorithm.
+         * @return the blob path for the path input.
+         */
+        public BlobPath path(PathInput pathInput, PathHashAlgorithm hashAlgorithm) {
+            DataCategory dataCategory = pathInput.dataCategory();
+            DataType dataType = pathInput.dataType();
+            assert dataCategory.isSupportedDataType(dataType) : "category:"
+                + dataCategory
+                + " type:"
+                + dataType
+                + " are not supported together";
+            return generatePath(pathInput, hashAlgorithm);
+        }
+
+        abstract BlobPath generatePath(PathInput pathInput, PathHashAlgorithm hashAlgorithm);
```

Comment:
The hashAlgorithm is not used in Fixed path, but otherwise it gets used. I am not in favour of having another method as that would lead to multiple if else ladder across 2 or more places. I am adding an assertion to ensure that hashAlgorithm is supplied as null for Fixed path usecase.